### ISSUE TITLE
gsWriteParaview: parametric mesh too detailed, thus taking a lot of time

### ISSUE THREAD

**Issue by dominik-mokris-mtu** [gismo/gismo#347]
We've had a discussion with @Sandra-Merchel-MTU recently about why `gsWriteParaview` was taking so much time on her computer. We have discovered the following problem.

When you call
```cpp
    gsWriteParaview(myTHB, "filename", 1000, true);
```
(because you want to see the parametric mesh), it will call `gsWriteSingleCompMesh` with `npts` equal to 1000, which is much more than the default 8. Consequently, you evaluate each edge of the parametric mesh in 1000 points, taking a lot of time (the THB face took from the MWE below took 20m, 59s on my computer). This does not seem necessary, as it results in the parametric mesh much finer than the rendering of the underlying geometry.

**MWE**
```cpp
#include <gismo.h>

using namespace gismo;

int main()
{
    gsFileData<real_t> data("surfaces/thbs_face_3levels.xml");
    gsTHBSpline<2>::uPtr THB = data.getFirst<gsTHBSpline<2> >();

    gsStopwatch time;
    time.restart();
    gsWriteParaview(*THB, "thb_out", 1000, true);
    time.stop();
    gsInfo << "gsWriteParaview took " << time << std::endl;
    return 0;
}
```
**Possible solutions**

1. We change the call of `gsWriteSingleCompMesh` to use `8` instead of `npts`.
2. We change the call of `gsWriteSingleCompMesh` to use some reasonable value derived from `npts`. Taking something of the order of `math::sqrt(mesh.numberOfElements() / npts)` should lead to a similar resolution of the parametric mesh and the underlying geometry.
3. We proclaim it a problem of the user. (-;

**Comment by dominik-mokris-mtu** [comment]
Good. I found one more thing I want to improve in `gsWriteParaview`. Once it's ready for a pull request, I will add this little enhancement there.

**Comment by dominik-mokris-mtu** [comment]
Solved in #349 , thanks to everyone who helped!
